package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -0,0 +1,943 @@
+                                            continue
+
+                                if zi == 2:
+                                    if self.coords_bbox[current_beadid, ZZ] + self.cutoff < self.grid.cellsize[ZZ] * (cellxyz[ZZ] + 1):
+                                        if self.coords_bbox[current_beadid, ZZ] + self.cutoff < self.box.c_pbcbox.box[ZZ][ZZ]:
+                                            # if self.debug and debug:
+                                            #     with gil:
+                                            #         print("FastNS:   -> Bead Z={:.3f}, Next cell Z={:.3f}, cutoff={:.3f} -> +Z shift is ignored".format(self.coords_bbox[current_beadid, ZZ], self.grid.cellsize[XX] * (cellxyz[ZZ] + 1), self.cutoff))
+                                            continue
+
+                            # Calculate and/or reinitialize shifted coordinates
+                            shifted_coord[XX] = self.coords[current_beadid, XX] + (xi - 1) * self.grid.cellsize[XX]
+                            shifted_coord[YY] = self.coords[current_beadid, YY] + (yi - 1) * self.grid.cellsize[YY]
+                            shifted_coord[ZZ] = self.coords[current_beadid, ZZ] + (zi - 1) * self.grid.cellsize[ZZ]
+
+                            probe[XX] = self.coords[current_beadid, XX] + (xi - 1) * self.cutoff
+                            probe[YY] = self.coords[current_beadid, YY] + (yi - 1) * self.cutoff
+                            probe[ZZ] = self.coords[current_beadid, ZZ] + (zi - 1) * self.cutoff
+
+                            # Make sure the shifted coordinates is inside the brick-shaped box
+                            for m in range(DIM - 1, -1, -1):
+
+                                while shifted_coord[m] < 0:
+                                    for d in range(m+1):
+                                        shifted_coord[d] += self.box.c_pbcbox.box[m][d]
+
+
+                                while shifted_coord[m] >= self.box.c_pbcbox.box[m][m]:
+                                    for d in range(m+1):
+                                        shifted_coord[d] -= self.box.c_pbcbox.box[m][d]
+
+                                while probe[m] < 0:
+                                    for d in range(m+1):
+                                        probe[d] += self.box.c_pbcbox.box[m][d]
+
+
+                                while probe[m] >= self.box.c_pbcbox.box[m][m]:
+                                    for d in range(m+1):
+                                        probe[d] -= self.box.c_pbcbox.box[m][d]
+
+                            # Get the cell index corresponding to the coord
+                            cellindex_adjacent = self.grid.coord2cellid(shifted_coord)
+                            cellindex_probe = self.grid.coord2cellid(probe)
+
+                            if cellindex == cellindex_probe and xi != 1 and yi != 1 and zi != 1:
+                                # if self.debug and debug:

```

Comment:
either remove all debug code or uncomment it.

Reply:
These comments are remains from some bug hunts... I will remove them